### DISCUSSION TITLE
Pinia action is not returning values

### DISCUSSION THREAD

**Discussion by SujithGunasekaran** [vuejs/pinia#1755] [Help and Questions] [answered]
Hi Team,

I have created test cases to test my code. Below is my .vue file code. Assume **getUserCart()** has an API call to get a user cart list from DB.

```
<script setup>
 onMounted(async () => {
  let result = await store.getUserCart();

  // while testing it returns undefined.
  console.log('test - 1', result);
 })
</script>
```

Below is my test case code.

```
it('should call the userCart', async () => {
    wrapper = shallowMount(UserCart, {
      global: {
        plugins: [
          i18n,
          router,
          store,
          createTestingPinia({ createSpy: vi.fn })
        ],
        stubs,
        provide: {
          '$heap': mocks.$heap
        },
      },
      props: {},
    });
    let userStore = useUserStore();
    userStore.userCartList = vi.fn().mockResolvedValueOnce(cartList);
    await flushPromises();
    await wrapper.vm.$nextTick();
});
```
If I remove **createTestingPinia** from the plunging and tried to run the test it is working. But when we provide **createTestingPinia** to plugin it is not working

When I debug the code store.getUserCart() returns undefined. Can someOne please help to correct if anything wrong in my code.

**Comment by posva** [comment, marked as answer]
Actions are stubbed: https://pinia.vuejs.org/cookbook/testing.html#customizing-behavior-of-actions

**Reply by SujithGunasekaran** [reply]
@posva Thanks for the response
